mixtcp/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-2.0-only
+
+mod bridge;
+mod device;
+mod error;
+
+pub use bridge::NymIprBridge;
+pub use device::NymIprDevice;
+pub use error::MixtcpError;
+
+use nym_ip_packet_requests::IpPair;
+use nym_sdk::stream_wrapper::IpMixStream;
+use tokio::sync::mpsc;
+
+/// Create a connected smoltcp device and async bridge for the tunneling packets through the
+/// Mixnet to remote hosts via an IPR.
+///
+/// This function handles the complete setup process:
+/// - Ensures the IPR stream is connected
+/// - Retrieves allocated IP addresses
+/// - Creates communication channels
+/// - Constructs the device and bridge components
+pub async fn create_device(
+    mut ipr_stream: IpMixStream,
```

Comment:
it almost feels as if mixtcp could have been its own independent crate without direct dependency on the sdk. could we perhaps abstract that `IpMixStream` away with some trait instead?